sympy/algebras/quaternion.py

```diff
@@ -754,10 +783,13 @@ def _eval_conjugate(self):
 
     def norm(self):
         """Returns the norm of the quaternion."""
-        q = self
-        # trigsimp is used to simplify sin(x)^2 + cos(x)^2 (these terms
-        # arise when from_axis_angle is used).
-        return sqrt(trigsimp(q.a**2 + q.b**2 + q.c**2 + q.d**2))
+        if self._norm is None:  # check if norm is pre-defined
+            q = self
+            # trigsimp is used to simplify sin(x)^2 + cos(x)^2 (these terms
```

Comment:
There is a fu helper to do that directly.

Reply:
What do you mean?

Reply:
![image](https://user-images.githubusercontent.com/90703/208462842-4ef7865a-e168-465e-8f5d-0d2ab7c128d4.png)

So a direct function rather than the more broad `simplify` can be used.

Reply:
Ok I see what you mean! 

I actually did not introduce `trigsimp` here, it was already part of the code. I only introduced the check for `_ norm` to keep the changes to a minimum... Don't you think this could be better introduced in a different PR?